src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfTests.cs

```diff
@@ -0,0 +1,707 @@
+                TestState = { Sources = { csInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
+            };
+            await testOrdinal.RunAsync();
+
+            operatorKind = operatorKind == " == " ? " = " : operatorKind;
+            notString = notString == "!" ? " Not" : notString;
+            string vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            Dim index As Integer = [|Str.IndexOf(""This"")|]
+            If index" + operatorKind + value + @" Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+
+            var testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(" == ", " -1", "!")]
+        [InlineData(" >= ", " 0", "")]
+        public async Task TestCharAndOrdinal(string operatorKind, string value, string notString)
+        {
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            const string str = ""This is a string"";
+            int index = [|str.IndexOf('a', System.StringComparison.Ordinal)|];
+            if (index" + operatorKind + value + @")
+            {
+
+            }
+        } 
+    } 
+}";
+            var testOrdinal = new VerifyCS.Test
+            {
+                TestState = { Sources = { csInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
+            };
+            await testOrdinal.RunAsync();
+
+            operatorKind = operatorKind == " == " ? " = " : operatorKind;
+            notString = notString == "!" ? " Not" : notString;
+            string vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            Dim index As Integer = [|Str.IndexOf(""a""c, System.StringComparison.Ordinal)|]
+            If index" + operatorKind + value + @" Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+
+            var testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("This", "This", false)]
+        [InlineData("a", "a", true)]
+        public async Task TestStringAndCharWithMultipleDiagnostics(string input, string fix, bool isCharTest)
```

Comment:
Consider adding test where local use is not immediately following the declaration, say:

```csharp
int index1 = [|str.IndexOf(" + startQuote + input + endQuote + @")|];
int index2 = [|str.IndexOf(" + startQuote + input + endQuote + @")|];
if (index1 == -1 || index2 == -1)
{
}
```

Reply:
Changed the unit test here like you suggested